storage/src/backend/rocksdb_backend.rs

```diff
@@ -43,101 +29,53 @@ pub struct RocksDBBackendStats {
     cache_total: u64,
 }
 
-impl StorageBackend for RocksDBBackend {
-    fn is_persisted(&self) -> bool {
-        true
-    }
-
-    fn put(&mut self, key: &EntryHash, value: ContextValue) -> Result<bool, StorageBackendError> {
-        let cf = self.inner.cf_handle(&self.column_name).ok_or(
-            StorageBackendError::MissingColumnFamily {
-                name: &self.column_name,
-            },
-        )?;
-
-        self.inner
-            .put_cf_opt(
-                cf,
-                &key.as_ref()[..],
-                &value,
-                &Self::default_write_options(),
-            )
-            .map_err(StorageBackendError::from)
-            .map(|_| true)
+impl GarbageCollector for RocksDBBackend {
+    fn new_cycle_started(&mut self) -> Result<(), StorageBackendError> {
+        Ok(())
     }
 
-    fn put_batch(
+    fn mark_reused(
         &mut self,
-        batch: Vec<(EntryHash, ContextValue)>,
+        _reused_keys: std::collections::HashSet<EntryHash>,
     ) -> Result<(), StorageBackendError> {
-        let mut rocksb_batch = WriteBatch::default(); // batch containing DB key values to persist
+        Ok(())
+    }
+}
+
+impl KeyValueStoreBackend<MerkleStorage> for RocksDBBackend {
+    fn is_persistent(&self) -> bool {
+        (self.inner.deref() as &dyn KeyValueStoreBackend<MerkleStorage>).is_persistent()
+    }
 
-        for (k, v) in batch.iter() {
-            (self.inner.deref() as &dyn KeyValueStoreWithSchema<MerkleStorage>).put_batch(
-                &mut rocksb_batch,
-                &k,
-                &v,
-            )?;
-        }
+    fn put(&self, key: &EntryHash, value: &ContextValue) -> Result<(), DBError> {
+        (self.inner.deref() as &dyn KeyValueStoreBackend<MerkleStorage>).put(key, value)
+    }
 
-        (self.inner.deref() as &dyn KeyValueStoreWithSchema<MerkleStorage>)
-            .write_batch(rocksb_batch)?;
+    fn delete(&self, key: &EntryHash) -> Result<(), DBError> {
+        (self.inner.deref() as &dyn KeyValueStoreBackend<MerkleStorage>).delete(key)
+    }
 
-        Ok(())
+    fn merge(&self, key: &EntryHash, value: &ContextValue) -> Result<(), DBError> {
+        (self.inner.deref() as &dyn KeyValueStoreBackend<MerkleStorage>).merge(key, value)
     }
 
-    fn merge(&mut self, key: &EntryHash, value: ContextValue) -> Result<(), StorageBackendError> {
-        let cf = self.inner.cf_handle(&self.column_name).ok_or(
-            StorageBackendError::MissingColumnFamily {
-                name: &self.column_name,
-            },
-        )?;
-
-        self.inner
-            .merge_cf_opt(
-                cf,
-                &key.as_ref()[..],
-                &value,
-                &Self::default_write_options(),
-            )
-            .map_err(StorageBackendError::from)
+    fn get(&self, key: &EntryHash) -> Result<Option<ContextValue>, DBError> {
+        (self.inner.deref() as &dyn KeyValueStoreBackend<MerkleStorage>).get(key)
```

Comment:
@mat13mn 
we almost never used deref() directly, I just saw few usages in merkle_storage stuff,
is not here any other solution without calling deref() ?

Reply:
i couldnt figure out anything else ;/

Reply:
Another thing about this that I find confusing, is that the methods on `inner` return `rocksdb::Error`, but `DBError` is expected here.

Reply:
Ok I think I now see why all this is needed: the `KeyValueStoreBackend` defines methods with the same names that RocksDB uses in `DB`, and that makes things ambiguous. I think `KeyValueStoreBackend` should be implemented directly over a wrapper over `DB` (newtype pattern), and not for `DB` itself (and probably good for the other trait implementations too).

What do you think?

Reply:
Thing is KeyValueStoreBackend is former `KeyValueStoreWithSchema` that is widely used in our codebase. One of the comment on previous PR was why to create separate trait(`StorageBackend`) as there is one already defined. What i did i splitted old `KeyValueStoreWithSchema` into 2 traits `KeyValueStoreBackend` and `KeyValueStoreWithSchemaIterator`. And created alias for old name 

```pub trait KeyValueStoreWithSchema<S: KeyValueSchema>:
    KeyValueStoreBackend<S> + KeyValueStoreWithSchemaIterator<S>
{                                        
}                                                   
```
to preserve backward compatibility. Then i just implemented already existing `KeyValueStoreBackend` for other types( inmem, btree, etc). I didnt want to mix old `DB` type with other merkle storage backends so i wrapped DB into RocksDBBackend and just forwarded all the method calls for that one. Does it make sense ?

Reply:
Yes, but there is still an implementation of that trait `impl<S: KeyValueSchema> KeyValueStoreWithSchemaIterator<S> for DB`, which is what I say I think is causing problems. If instead of implementing anything `for DB` you instead create a new wrapper type and move that implementation over there (so that it is not implemented `for DB` anymore), the ambiguity (which is what I think is causing problems) goes away.

Reply:
That applies to all traits we implement `for DB`.

Reply:
added todo note TE-437